### PR TITLE
Default npm algorithm

### PR DESCRIPTION
**Pull request by pYr0x**
set the default npm algorithm to "flat" (isFlatFileStructure=true)
close https://github.com/stealjs/steal/issues/701

### CODE REVIEW

**Comment by pYr0x** [comment]
i meant "update travis CI TO run on node 4 and 5.5 ;)

@matthewp let me know if this PR is OK

File: npm.js [hunk 1]
```diff
@@ -41,13 +41,21 @@ exports.translate = function(load){
...
-	if(pkg.system && pkg.system.npmAlgorithm === "flat") {
-		context.isFlatFileStructure = true;
+
+	var pkgName = pkg.name;
+	// backwards compatible for < npm 3
+	if(pkg.system && pkg.system.npmAlgorithm === "nested") {
+		context.isFlatFileStructure = false;
+	} else {
+		pkg.system = pkg.system || {};
+		pkg.system.npmAlgorithm = "flat";
```
Comment (matthewp) [inline npm.js]: Is this part needed? I think setting isFlatFileStructure to true (which you did above) is enough.

File: npm.js [hunk 2]
```diff
@@ -59,6 +67,10 @@ exports.translate = function(load){
 				if(pkg.browser){
 					delete pkg.browser.transform;
 				}
+				if(pkg.system && pkg.system.npmAlgorithm === "nested" && context.isFlatFileStructure === true) {
```
Comment (matthewp) [inline npm.js]: I don't think we should warn here. npmAlgorithm is essentially a dev-mode option. If a library author is using `nested` because they themselves are developing with npm2, that doesn't mean it won't work with npm3.  Only the project that is running `npm install` matters.

File: test/helpers.js [hunk 3]
```diff
@@ -153,19 +153,20 @@ Runner.prototype.npmVersion = function(version){
 	if(arguments.length === 0) {
 		return this._version; 
 	}
-
-	this.algorithm = version >= 3 ? "flat": undefined;
 	this._version = version;
-	this._addVersion();
```
Comment (matthewp) [inline test/helpers.js]: Why was this removed?  Now `_addVersion` is only called when rootPackage is called, so if you did something like:

```js
helpers.clone()
  .rootPackage({
     ...
  })
  .npmVersion(2)
```

_addVersion wouldn't run.

Comment (pYr0x) [inline test/helpers.js]: you shouldn't set a NPM version by using `.npmVersion()`. 
the old `.npmVersion()`. overwrites the `npmAlgoritm` but not the `isFlatFileStructure`
the way we check the npm version is by scanning the npmAlgoritm in `package.system`.

so if you want to using npm < 3 you have to set a `nested` property in `package.system` like
https://github.com/stealjs/system-npm/pull/146/files#diff-e9a7e88b9a1c4fff3b21fb49ed79b8d7R104

File: npm.js (hunk 1 above)
Comment (pYr0x) [inline npm.js]: if we dont set `pkg.system.npmAlgorithm = "flat";` the `System.npmAlgorithm` attribute is `undefined` by default. 

which is like now if no `npmAlgorithm` is defined in package.json
i thought you want npmAlgorithm = flat 
https://github.com/stealjs/steal/issues/701

File: npm.js (hunk 2 above)
Comment (pYr0x) [inline npm.js]: can we put a `//!steal-remove-start` so in production this is not being a part?

File: npm.js (hunk 1 above)
Comment (matthewp) [inline npm.js]: I don't so much care about `System.npmAlgorithm`, it's not something you would use in an application... but I guess we can leave it just in case there's some weird use-case where a user needs to know which npm version is used.

File: npm.js (hunk 2 above)
Comment (matthewp) [inline npm.js]: Honestly, I think we should remove it. I find too many warnings causes people to not read the important ones. In this case I can't think of any scenario where this would be useful...

File: test/helpers.js (hunk 3 above)
Comment (matthewp) [inline test/helpers.js]: Yeah, I sort of agree, should set the algorithm with package.system in `rootPackage`, the helper sort of obscures what is going on, so I'm ok with getting rid of it.

File: npm.js (hunk 2 above)
Comment (pYr0x) [inline npm.js]: ok i will remove it

**Comment by pYr0x** [comment]
close this for a new PR against major branch
